windows/setup.ps1

```diff
@@ -28,6 +28,29 @@ $result = New-Object psobject @{
 $win32_os = Get-CimInstance Win32_OperatingSystem
 $win32_cs = Get-CimInstance Win32_ComputerSystem
 $osversion = [Environment]::OSVersion
+
+### ADD VIRTUAL PRODUCT INFO ###
+if (((Get-WmiObject win32_bios).Version).tolower().contains("version")) {
+        $virtualization_type = "Hyper-V"
+}
+if (((Get-WmiObject win32_bios).SerialNumber).tolower().contains("vmware")) {
+        $virtualization_type = "VMWare"
+}
+if (((Get-WmiObject win32_bios).Version).tolower().contains("xen")) {
+        $virtualization_type = "Xen"
+}
+if (((Get-WmiObject win32_bios).Version).tolower().contains("vbox")) {
+        $virtualization_type = "VirtualBox"
+}
```

Comment:
This isn't quite robust enough - have a look at https://blogs.technet.microsoft.com/heyscriptingguy/2015/03/30/use-powershell-to-detect-if-hypervisor-is-present/ and http://blogs.splunk.com/2013/08/12/detecting-your-hypervisor-from-within-a-windows-guest-os/ as starting points. I can see a number of cases where this code will false-trigger Hyper-V for real BIOSes. Need to figure out if we're under a Hypervisor at all first, then try to identify...